src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
Thank your for all of this great input, and spending some time with me on this. Very interesting approach. I'm plugged in this shortened code to play with it, but the result isn't coming out quite right. The signature for the overridden `ap()` method isn't what you wrote above, but `<B> Applicative<Validation<E, ?>, B> ap(Kind<Validation<E, ?>, ? extends Function<? super T, ? extends B>> f)`. If I force it to `<B> Validation<E, B> ap(Validation<E, ? extends Function<? super T, ? extends B>> f)`, that signature works, but doesn't work as the overridden  `Applicative` method `ap()`.

I'm still struggling with whether this is all worth it. This is only worth all this work if we feel we could create other Applicative types. If not, then we should go the simpler route for now and keep all our implemented code in `Validation`, which I already have a working version of. I'm curious about your thoughts on all of this and if the complexity is worth it? Thanks so much for your help on this Daniel.